proto/desmos/reports/v1/query.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+package desmos.reports.v1;
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+import "cosmos_proto/cosmos.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+
+import "desmos/reports/v1/models.proto";
+
+option go_package = "github.com/desmos-labs/desmos/v3/x/reports/types";
+
+// Query defines the gRPC querier service.
+service Query {
+  // Reports allows to query the reports for a specific target
+  rpc Reports(QueryReportsRequest) returns (QueryReportsResponse) {
```

Comment:
Currently, to know if the user reported to the a target post or target user is required to progress as follows:
1. Get all the reports from `Reports` query
2. Search all reports whose reportor is the user via pagination

If there are millions of reports on a post, then the check to the report id for a user would be costly.
Therefore, I think we should have a `UserReport` query to improve the performance.

Reply:
Since implementing this method would require a more complex logic and additional store keys, I don't think we should have it. All these kinds of queries can easily be implemented using an external data parser like [DJuno](https://github.com/desmos-labs/djuno) that can expose a GraphQL endpoint anyway

Reply:
Okay, I found that the query is not useful since we don't limit the user to report a post or a user multiple times. 
Shouldn't we check the user has reported yet when creating a report?

Reply:
Initially I didn't check for duplicated reports due to the fact that a single report could be associated to a single reason. So, if you wanted to report a post for two reasons (e.g. spamming and inadequate contents) you would have to create two different reports. I've now changed this to allow a single report to be associated to multiple reasons. This allows to check for duplicated reports but requires us to remove the reporting reason ids from the events inside `MsgCreateReport` due to the fact that arrays cannot be indexed properly (I think it's fine anyway)